luaui/Widgets/gfx_lupsgl4_orbs.lua

```diff
@@ -0,0 +1,748 @@
+		return mix(x1, x2, xy.y);
+	}
+
+	float noise13( vec3 P ) {
+		//  https://github.com/BrianSharpe/Wombat/blob/master/Perlin3D.glsl
+
+		// establish our grid cell and unit position
+		vec3 Pi = floor(P);
+		vec3 Pf = P - Pi;
+		vec3 Pf_min1 = Pf - 1.0;
+
+		// clamp the domain
+		Pi.xyz = Pi.xyz - floor(Pi.xyz * ( 1.0 / 69.0 )) * 69.0;
+		vec3 Pi_inc1 = step( Pi, vec3( 69.0 - 1.5 ) ) * ( Pi + 1.0 );
+
+		// calculate the hash
+		vec4 Pt = vec4( Pi.xy, Pi_inc1.xy ) + vec2( 50.0, 161.0 ).xyxy;
+		Pt *= Pt;
+		Pt = Pt.xzxz * Pt.yyww;
+		const vec3 SOMELARGEFLOATS = vec3( 635.298681, 682.357502, 668.926525 );
+		const vec3 ZINC = vec3( 48.500388, 65.294118, 63.934599 );
+		vec3 lowz_mod = vec3( 1.0 / ( SOMELARGEFLOATS + Pi.zzz * ZINC ) );
+		vec3 highz_mod = vec3( 1.0 / ( SOMELARGEFLOATS + Pi_inc1.zzz * ZINC ) );
+		vec4 hashx0 = fract( Pt * lowz_mod.xxxx );
+		vec4 hashx1 = fract( Pt * highz_mod.xxxx );
+		vec4 hashy0 = fract( Pt * lowz_mod.yyyy );
+		vec4 hashy1 = fract( Pt * highz_mod.yyyy );
+		vec4 hashz0 = fract( Pt * lowz_mod.zzzz );
+		vec4 hashz1 = fract( Pt * highz_mod.zzzz );
+
+		// calculate the gradients
+		vec4 grad_x0 = hashx0 - 0.49999;
+		vec4 grad_y0 = hashy0 - 0.49999;
+		vec4 grad_z0 = hashz0 - 0.49999;
+		vec4 grad_x1 = hashx1 - 0.49999;
+		vec4 grad_y1 = hashy1 - 0.49999;
+		vec4 grad_z1 = hashz1 - 0.49999;
+		vec4 grad_results_0 = inversesqrt( grad_x0 * grad_x0 + grad_y0 * grad_y0 + grad_z0 * grad_z0 ) * ( vec2( Pf.x, Pf_min1.x ).xyxy * grad_x0 + vec2( Pf.y, Pf_min1.y ).xxyy * grad_y0 + Pf.zzzz * grad_z0 );
+		vec4 grad_results_1 = inversesqrt( grad_x1 * grad_x1 + grad_y1 * grad_y1 + grad_z1 * grad_z1 ) * ( vec2( Pf.x, Pf_min1.x ).xyxy * grad_x1 + vec2( Pf.y, Pf_min1.y ).xxyy * grad_y1 + Pf_min1.zzzz * grad_z1 );
+
+		// Classic Perlin Interpolation
+		vec3 blend = Pf * Pf * Pf * (Pf * (Pf * 6.0 - 15.0) + 10.0);
+		vec4 res0 = mix( grad_results_0, grad_results_1, blend.z );
+		vec4 blend2 = vec4( blend.xy, vec2( 1.0 - blend.xy ) );
+		float final = dot( res0, blend2.zxzx * blend2.wwyy );
+		return ( final * 1.1547005383792515290182975610039 );  // scale things to a strict -1.0->1.0 range  *= 1.0/sqrt(0.75)
+	}
+
+	float Fbm12(vec2 P) {
+		const int octaves = 2;
+		const float lacunarity = 1.5;
+		const float gain = 0.49;
+
+		float sum = 0.0;
+		float amp = 1.0;
+		vec2 pp = P;
+
+		int i;
+
+		for(i = 0; i < octaves; ++i)
+		{
+			amp *= gain;
+			sum += amp * noise12(pp);
+			pp *= lacunarity;
+		}
+		return sum;
+	}
+
+	float Fbm31Magic(vec3 p) {
+		 float v = 0.0;
+		 v += noise13(p * 1.0) * 2.200;
+		 v -= noise13(p * 4.0) * 3.125;
+		 return v;
+	}
+
+	float Fbm31Electro(vec3 p) {
+		 float v = 0.0;
+		 v += noise13(p * 0.9) * 0.99;
+		 v += noise13(p * 3.99) * 0.49;
+		 v += noise13(p * 8.01) * 0.249;
+		 v += noise13(p * 15.05) * 0.124;
+		 return v;
+	}
+
+	#define SNORM2NORM(value) (value * 0.5 + 0.5)
+	#define NORM2SNORM(value) (value * 2.0 - 1.0)
+
+	#define time (gameFrame_vs * 0.03333333)
```

Comment:
(requires the change above from x+w to z)
```suggestion
	#define time gameFrame_vs
```
or perhaps even this and remove `gameFrame_vs`?

```suggestion
	#define time timeInfo.z
```